InventoryManagementSystem/Domain/InventoryManagement/Inventory.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using InventoryManagementSystem.Domain.Reader;
+
+namespace InventoryManagementSystem.Domain
+{
+    public class Inventory
+    {
+        public List<Product> products = new List<Product>();
+        private readonly IReader reader;
+
+        public Inventory(IReader reader)
+        {
+           this.reader = reader;
+        }
+        public void AddProduct()
+        {
+            Product product = reader.ReadProduct();
+            products.Add(product);
+        }
+        public void ViewAllProducts()
+        {
+            Console.WriteLine("The Products:");
+            foreach (var product in products)
+            {                                  
+                Console.WriteLine(product);
+            }
+        }
+        public void EditProduct()
+        {
+            string? name = reader.ReadString("Enter the name of the product You want To Edit:");
+            Product? EditedProduct = products.Find(x => x.Name == name);
+            if (EditedProduct == null)
+            {
+                Console.WriteLine("Product not found in inventory.");
+                return;
+            }
+            foreach (var product in products)
+            {
+            if (EditedProduct.Name == product.Name)
+                {
+                    string? newName = reader.ReadString("Enter the new Product Name:");
+                    EditedProduct.Name = newName;
+
+                    int newprice = reader.ReadInt("Enter the new Product Price:");
+                    EditedProduct.Price = newprice;
+
+                    int newQuantity = reader.ReadInt("Enter the quantity of the product");
+                    EditedProduct.Quantity = newQuantity;
```

Comment:
is not it better to read all these fields as a product instead of reading them individually?, I mean the IReader already has a ReadProduct method, why not to use it?